src/plugins/discover/public/application/angular/doc_table/components/table_row.tsx

```diff
@@ -0,0 +1,258 @@
+import { DOC_HIDE_TIME_COLUMN_SETTING } from '../../../../../common';
+import { TableCell } from './table_row/table_cell';
+import { formatRow, formatTopLevelObject } from '../../helpers';
+import { getContextUrl } from '../../../helpers/get_context_url';
+import { DocViewer } from '../../../components/doc_viewer/doc_viewer';
+import { DocViewFilterFn, ElasticSearchHit } from '../../../doc_views/doc_views_types';
+
+// guesstimate at the minimum number of chars wide cells in the table should be
+const MIN_LINE_LENGTH = 20;
+
+export type DocTableRow = ElasticSearchHit & {
+  isAnchor?: boolean;
+};
+
+export interface TableRowProps {
+  columns: string[];
+  filter: DocViewFilterFn;
+  indexPattern: IndexPattern;
+  row: DocTableRow;
+  onAddColumn?: (column: string) => void;
+  onRemoveColumn?: (column: string) => void;
+  useNewFieldsApi: boolean;
+}
+
+export const TableRow = ({
+  columns,
+  filter,
+  row,
+  indexPattern,
+  useNewFieldsApi,
+  onAddColumn,
+  onRemoveColumn,
+}: TableRowProps) => {
+  const services = getServices();
+
+  const [open, setOpen] = useState(false);
+  const docTableRowClassName = classNames('kbnDocTable__row', {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    'kbnDocTable__row--highlight': row.isAnchor,
+  });
+  const anchorDocTableRowSubj = row.isAnchor ? ' docTableAnchorRow' : '';
+
+  const flattenedRow = useMemo(() => indexPattern.flattenHit(row), [indexPattern, row]);
+  const mapping = useMemo(() => indexPattern.fields.getByName, [indexPattern]);
+  const hideTimeColumn = useMemo(
+    () => services.uiSettings.get(DOC_HIDE_TIME_COLUMN_SETTING, false),
+    [services]
+  );
+
+  // toggle display of the rows details, a full list of the fields from each row
+  const toggleRow = () => setOpen((prevOpen) => !prevOpen);
+
+  /**
+   * Fill an element with the value of a field
+   */
+  const _displayField = useCallback(
+    (fieldName: string, truncate = false) => {
+      const text = indexPattern.formatField(row, fieldName);
+      // eslint-disable-next-line react/no-danger
+      const fieldElement = <span dangerouslySetInnerHTML={{ __html: text }} />;
+
+      if (truncate && text.length > MIN_LINE_LENGTH) {
+        return <div className="truncate-by-height">{fieldElement}</div>;
+      }
+
+      return fieldElement;
+    },
+    [indexPattern, row]
+  );
+
+  const inlineFilter = (column: string, type: '+' | '-') => {
+    const field = indexPattern.fields.getByName(column);
+    filter(field!, flattenedRow[column], type);
+  };
+
+  const getContextAppHref = () => {
+    return getContextUrl(
+      row._id,
+      indexPattern.id!,
+      columns,
+      services.filterManager,
+      services.addBasePath
+    );
+  };
+
+  const getSingleDocHref = () => {
+    return services.addBasePath(
+      `/app/discover#/doc/${indexPattern.id!}/${row._index}?id=${encodeURIComponent(row._id)}`
+    );
+  };
+
+  const newHtmls = [
```

Comment:
:art: Not sure I understand the varibale naming here, why are we calling this `newHtmls`?

Reply:
Renamed